### PR TITLE
addOwner function now requires the new owner to not be an owner

### LINKED ISSUE #63: `IdentityManager::addOwner()` can be called for existing owner
https://github.com/uport-project/uport-identity/blob/880f8b81b9355edfe62c35817562b24e90513d2a/contracts/IdentityManager.sol#L116-L119

As either `addOwner()` function is written, a malicious olderOwner could call addOwner on an existing owner of `owner` or `olderOwner` status. This function call [overwrites](https://github.com/uport-project/uport-identity/blob/880f8b81b9355edfe62c35817562b24e90513d2a/contracts/IdentityManager.sol#L117) the time that existing owner became an owner in the `owners` mapping, and may be used as an attack to remove an older owner's admin privileges.

**Recommendation**

We recommend the addition of a `require` statement in both functions that ensures that `addOwner()` can only be called on an owner that is not a current owner.

### LINKED ISSUE #64: `MetaIdentityManager::addOwner()` can be called for existing owner
https://github.com/uport-project/uport-identity/blob/259435e45fc81167d137b33d40215a796b1cc091/contracts/MetaIdentityManager.sol#L127-L130

As either `addOwner()` function is written, a malicious olderOwner could call addOwner on an existing owner of `owner` or `olderOwner` status. This function call [overwrites](https://github.com/uport-project/uport-identity/blob/880f8b81b9355edfe62c35817562b24e90513d2a/contracts/IdentityManager.sol#L117) the time that existing owner became an owner in the `owners` mapping, and may be used as an attack to remove an older owner's admin privileges.

**Recommendation**

We recommend the addition of a `require` statement in both functions that ensures that `addOwner()` can only be called on an owner that is not a current owner.

### PR DESCRIPTION
**Pull request by oed**
fixes #63
fixes #64

[ch4720]
[ch4721]

### CODE REVIEW

File: contracts/IdentityManager.sol [hunk 1]
```diff
@@ -114,6 +114,7 @@ contract IdentityManager {
 
     /// @dev Allows an olderOwner to add a new owner instantly
     function addOwner(Proxy identity, address newOwner) public onlyOlderOwner(identity) rateLimited(identity) {
+        require(!isOwner(identity, newOwner));
```
Comment (maurelian) [inline contracts/IdentityManager.sol]: LGTM

File: contracts/MetaIdentityManager.sol [hunk 2]
```diff
@@ -129,6 +129,7 @@ contract MetaIdentityManager {
         onlyOlderOwner(identity, sender)
         rateLimited(identity, sender)
     {
+        require(!isOwner(identity, newOwner));
```
Comment (maurelian) [inline contracts/MetaIdentityManager.sol]: LGTM

File: test/identityManager.js [hunk 3]
```diff
@@ -145,6 +145,16 @@ contract('IdentityManager', (accounts) => {
       assert.isTrue(isOwner, 'user5 should be owner now')
     })
 
+    it('owner can not add owner that is already an owner', async function() {
+      try {
+        let tx = await identityManager.addOwner(proxy.address, user1, {from: user1})
```
Comment (maurelian) [inline test/identityManager.js]: I don't think this should be user1 trying to add user1.
I know it's still an owner trying to add an existing owner, but this is a subset edge case of the condition being tested.

File: test/metaIdentityManager.js [hunk 4]
```diff
@@ -183,6 +183,16 @@ contract('MetaIdentityManager', (accounts) => {
       assert.isTrue(isOwner, 'user5 should be owner now')
     })
 
+    it('owner can not add owner that is already an owner', async function() {
+      try {
+        let tx = await identityManager.addOwner(user1, proxy.address, user1, {from: user1})
```
Comment (maurelian) [inline test/metaIdentityManager.js]: Same as comment on previous test edits.
